### PR TITLE
feat: calculate virtual elements for CDS Views

### LINKED ISSUE #12: CDS View -Virtual Elements empty
When selecting data from a CDS View which contains virtual elements, the columns stay empty in the result table.

Execute the following steps to reproduce the issue

1. Start DB Browser
2. Choose Browser Mode: "CDS View"
3. Enter a CDS view with virtual elements
4. Execute the search

The result list contains the virtual element as a column, but it is always empty.

### PR DESCRIPTION
**Pull request by reichr-dev**

### CODE REVIEW

**Comment by reichr-dev** [comment]
Resolves Issue #12

File: src/ui/util/zcl_dbbr_cds_selection_util.clas.abap [hunk 1]
```diff
@@ -581,4 +596,78 @@ CLASS zcl_dbbr_cds_selection_util IMPLEMENTATION.
...
+            EXPORTING
+              it_original_data           = <lt_selected_data>
+              it_requested_calc_elements = VALUE if_sadl_exit_calc_element_read=>tt_elements( ( CONV #( <ls_virtual_elem>-fieldname ) ) )
+            CHANGING
+              ct_calculated_data         = <lt_calculated_data>.
+
+        CATCH cx_sy_create_object_error
+            cx_sy_ref_is_initial
+            cx_sy_dyn_call_error
+            cx_sadl_exit.
```
Comment (stockbal) [inline src/ui/util/zcl_dbbr_cds_selection_util.clas.abap]: the exception `cx_sadl_exit` does not exist in v7.50 so I gues we have to use `cx_root` in this case to also catch any exception that happen in the `calculate` call

File: src/ui/util/zcl_dbbr_cds_selection_util.clas.abap [hunk 2]
```diff
@@ -581,4 +596,78 @@ CLASS zcl_dbbr_cds_selection_util IMPLEMENTATION.
...
+          RETURN.
+      ENDTRY.
+
+    ENDLOOP.
```
Comment (stockbal) [inline src/ui/util/zcl_dbbr_cds_selection_util.clas.abap]: I am not quite sure how the `SADL` engine is handling this, but wouldn't it be better to group the calculcation by exit-class?

Example:
```ABAP
...
@ObjectModel.virtualElementCalculatedBy: 'ABAP:CL_EXIT'
cast( '' as ztype ) as VirtualField1,

@ObjectModel.virtualElementCalculatedBy: 'ABAP:CL_EXIT'
cast( '' as ztype2 ) as VirtualField2,

@ObjectModel.virtualElementCalculatedBy: 'ABAP:CL_EXIT2'
cast( '' as ztype3 ) as VirtualField3,
...
```
So in this example there would be one call to `CL_EXIT` where the fields `VirtualField1` and `VirtualField2` are filled and one call to `CL_EXIT2` where the field `VirtualField3` is filled

File: src/ui/util/zcl_dbbr_cds_selection_util.clas.abap [hunk 3]
```diff
@@ -581,4 +596,78 @@ CLASS zcl_dbbr_cds_selection_util IMPLEMENTATION.
...
+
+      TRY.
+          CREATE OBJECT lo_calculated_by TYPE (<ls_virtual_elem>-value+5).
+
+          create_virtual_elem_table_type(
+            EXPORTING iv_virtual_elem = <ls_virtual_elem>-fieldname
+            IMPORTING eo_table_type = DATA(lo_table_type) ).
+
+          CREATE DATA lr_t_calc_data TYPE HANDLE lo_table_type.
+          ASSIGN lr_t_calc_data->* TO <lt_calculated_data>.
```
Comment (stockbal) [inline src/ui/util/zcl_dbbr_cds_selection_util.clas.abap]: Wouldn't this automatically override the calculated data from the previous iteration?

**Review by stockbal — CHANGES_REQUESTED** [review]
I guess all in all the coding assumes that there is only one calculated field in the view. Then it would work

**Pull request closed without merging**